node-definitions/freebsd/README.md

```diff
@@ -0,0 +1,58 @@
+# FreeBSD in CML
+
+Instructions for running a basic FreeBSD node in Cisco Modeling Lab.
+
+# Deployable image
+
+A pre-built FreeBSD image is available
+[here](https://download.freebsd.org/snapshots/VM-IMAGES/14.1-STABLE/amd64/Latest/FreeBSD-14.1-STABLE-amd64-BASIC-CLOUDINIT.ufs.qcow2.xz).
+
+In Linux the above image can be decomressed with:
+
+    xz --decompress FreeBSD-14.1-STABLE-amd64-BASIC-CLOUDINIT.ufs.qcow2.xz
+
+From there the qcow2 file can be imported into CML.
+
+By default the FreeBSD image is configured to use DHCP.  It also supports a
+native serial console which is ideal for CML.
+
+# FreeBSD node definition
+
+A FreeBSD [node definition](FreeBSD.yaml) is include in this directory.
+
+# Limitations and future work
+
+The FreeBSD 14.1 "Basic Cloud-init" image comes with "nuage-init" rather than
+"cloud-init".  Nuage-init does not support all options that are available in
+cloud-init.  A base configuration is provided in this
+[node definition](FreeBSD.yaml).
+
+If the CML node has Internet connectivity it is possible to use the package
+manager to install cloud-init.  This will allow you to use cloud-init on
+subsequent boots.  This approach may have limited utility if you are using
+Terraform to provision the CML node.
+
+## Nuage-init: MAC address required
+
+If you are attempting to use Nuage-init in order to set an IP address please
+have a look at the [source code](https://github.com/freebsd/freebsd-src/blob/f35ccf46c7c6cb7d26994ec5dc5926ea53be0116/libexec/nuageinit/nuageinit#L283-L293).
+
+Nuage-init will explicitly check to see that there is an ```ethernets```
+section with a ```match``` subsection.  The only working match clause is the
+```macaddress``` clause.  If you do not explicitly state the MAC address of
```

Comment:
Not a knock on you, but this is **lame**.  The good news is in CML 2.8 you'll be able to customize MAC addresses.

Reply:
Agreed.  I saw the news about CML 2.8 about 20 minutes after I submitted the PR.  It gave me hope.